### ISSUE TITLE
`self-assigning-variable` (`PLW0127`) false negative on multi assignment

### ISSUE THREAD

**Issue by DetachHead** [astral-sh/ruff#8667]
```py
a = 1
a = a # error

b = b = 1 # no error
```
[playground](https://play.ruff.rs/71ab81ea-2846-4e7e-9c22-b1978ed84ace)

**Comment by charliermarsh** [comment]
Should the issue title be "false negative"? I want to make sure I understand the issue.

**Comment by DetachHead** [comment]
yeah my bad

**Comment by charliermarsh** [comment]
No worries, thanks!

**Comment by DetachHead** [comment]
if this is a https://github.com/astral-sh/ruff/labels/good%20first%20issue i'll have a go at fixing it myself
